aimmo-game/tests/functional/test_health_pickups_and_effects.py

```diff
@@ -1,44 +1,53 @@
-import random
 from unittest import TestCase
 
-from tests.test_simulation.dummy_avatar import DummyAvatarManager, MoveEastDummy
-from tests.test_simulation.dummy_avatar import MoveEastDummy
+from mock_world import MockWorld
 
-from simulation import map_generator
-from simulation.turn_manager import SequentialTurnManager
 from simulation.location import Location
 from simulation.pickups import HealthPickup
 
 
-SETTINGS = {
-    'START_HEIGHT': 5,
-    'START_WIDTH': 5,
-    'OBSTACLE_RATIO': 0,
-}
+class TestHealthPickupAndEffects(TestCase):
+    def test_health_pickups_and_effects_apply_default(self):
+        game = MockWorld()
+        game.game_state.add_avatar(1, None, Location(0, 0))
+        cell = game.game_state.world_map.get_cell(Location(0, 0))
+        self.assertEqual(cell.avatar, game.avatar_manager.get_avatar(1))
+        self.assertEqual(cell.avatar.health, 5)
 
+        cell = game.game_state.world_map.get_cell(Location(1, 0))
+        cell.pickup = HealthPickup(cell)
 
-class TestGame(object):
-    def __init__(self, settings=SETTINGS):
-        random.seed(0)
-        self.generator = map_generator.Main(settings)
-        self.avatar_manager = DummyAvatarManager([MoveEastDummy])
-        self.game_state = self.generator.get_game_state(self.avatar_manager)
-        self.turn_manager = SequentialTurnManager(game_state=self.game_state, end_turn_callback=lambda: None,
-                                                  completion_url='')
+        game.turn_manager._run_single_turn()
 
+        self.assertEqual(cell.avatar, game.avatar_manager.get_avatar(1))
+        self.assertEqual(cell.avatar.health, 8)
 
-class TestHealthPickupAndEffects(TestCase):
-    def test_health_pickups_and_effects_apply(self):
-        game = TestGame()
+    def test_health_pickups_and_effects_apply_custom(self):
+        game = MockWorld()
         game.game_state.add_avatar(1, None, Location(0, 0))
         cell = game.game_state.world_map.get_cell(Location(0, 0))
         self.assertEqual(cell.avatar, game.avatar_manager.get_avatar(1))
         self.assertEqual(cell.avatar.health, 5)
 
         cell = game.game_state.world_map.get_cell(Location(1, 0))
-        cell.pickup = HealthPickup(cell)
+        cell.pickup = HealthPickup(cell, 10.5)
 
         game.turn_manager._run_single_turn()
 
         self.assertEqual(cell.avatar, game.avatar_manager.get_avatar(1))
-        self.assertEqual(cell.avatar.health, 8)
+        self.assertEqual(cell.avatar.health, 16)
+
+    def test_health_effect_is_capped_at_100(self):
+        game = MockWorld()
+        game.game_state.add_avatar(1, None, Location(0, 0))
+        cell = game.game_state.world_map.get_cell(Location(0, 0))
+        self.assertEqual(cell.avatar, game.avatar_manager.get_avatar(1))
+        self.assertEqual(cell.avatar.health, 5)
+
+        cell = game.game_state.world_map.get_cell(Location(1, 0))
+        cell.pickup = HealthPickup(cell, 100)
```

Comment:
do we want it to be so hardcoded and things?

Reply:
The 100? It's a test, hardcoding will mean we are sure of what outcome we want, no? Unless I misunderstood?

Reply:
But we could definitely use hypothesis here ====)

Reply:
Might be good to have settings for this value instead of 100. Like: MAX_HEALTH or whatever. Which should be used in the code too, for another refactoring pr :) I think it would make things easier